### ISSUE TITLE
check_source: has_whitelist_warnings(): no log file fairly often

### ISSUE THREAD

**Issue by jberry-suse** [openSUSE/openSUSE-release-tools#2192]
Five occurrences in the last 24 hours:

```
devel:openSUSE:Factory/standard/i586/libica
devel:openSUSE:Factory/standard/x86_64/libica
X11:xfce/images/i586/package-update-indicator
X11:xfce/images/x86_64/package-update-indicator
Education/images/local/maxima
```

https://github.com/openSUSE/openSUSE-release-tools/blob/79a13dddfe8312d2ecf0adf411b663332a05d127/check_source.py#L233-L241

It might make sense to improve this to wait for a least one arch to build (or one of the target project archs) instead of just skip over a fairly to find log. Given this behavior one should be able to constantly trigger rebuilds and void this check.

This would not exist if part of staging process (or waits for those build results) which is related @DimStar77's comment about utilizing project config to trigger rpmlint errors.

**Comment by jsegitz** [comment]
yes that is a risk (I was aware of it when I implemented it). Waiting for the build results for one arch would allow users to stall the bot indefinitely, which is also problematic.

One idea would be to wait for a given time and fail if no build result is found. But I fear that this would cause issues when the build service is under high load.

**Comment by jberry-suse** [comment]
Depending on staging would be a sure thing, but mixes responsibilities a bit since repo-checker operated as a reviewbot and no longer does on staging projects. Otherwise, at minimum this should be downgraded from error since expected. As for stalling, it would merely stall the review on a request tied to package being stalled and not the entire load of requests.

**Comment by jberry-suse** [comment]
![image](https://user-images.githubusercontent.com/22943929/64292850-e9d5e600-cf30-11e9-917c-8f3b8aad7f91.png)

**Comment by coolo** [comment]
Shouldn't we check the build results in stagings anyway?

**Comment by jberry-suse** [comment]
> Shouldn't we check the build results in stagings anyway?

Yes. Just means factory-auto will block. At that point this likely makes more sense to be done as a staging check like repo-checker.

**Comment by coolo** [comment]
Having factory-auto wait for staging is pretty much recursive and doesn't really make sense. So we need a reviewer that is added to SRs and then triggered from staging-installcheck. Unlike repo checker we don't need to rerun this check - we only need the build log once.

**Comment by jsegitz** [comment]
We want to turn the errors into warnings for all rpmlint instances. Therefor we need a solution that works for projects other than openSUSE as well and we don't have the staging mechanism there. I'm beginning to doubt that this is the correct approach to the problem and will talk with my team about this. Thank you for your input, I'll get back here on Monday

**Comment by jberry-suse** [comment]
Can make a condition for "if staging" then utilize those packages, but either way something should be changed. Please reduce log message to info if going to leave this as expected.

**Comment by jsegitz** [comment]
So I came up with
https://github.com/jsegitz/openSUSE-release-tools/commit/c6cef31d7984619c1ee31991b8341bd3350b462b
This still gives us a few false positives. I'm also trying to solve this directly in rpmlint, but I'm interested if the timeout is something that would be acceptable to you

**Comment by coolo** [comment]
As mentioned before: for factory any build result from the devel project is unacceptable. You'll never know if the build log you look at is actually for the sources reviewed.

**Comment by coolo** [comment]
I'm tracking this as bug as I understand the security team thinks that the whitelist is tracked correctly - which isn't the case for 2 1/2 years. But there is noone there to fix it, so all I can do is revert it in total

**Comment by coolo** [comment]
So I would revert this from check_source and add that into staging-installcheck. There we have reliable build logs - and we still could add security reviews (and ignore the package if present)

**Comment by jsegitz** [comment]
For us this is a "safety net", we don't fully rely on this. If we have reliable build logs in staging-installcheck then this of course would be better for us. Do you need help with that?

**Comment by coolo** [comment]
I guess not
